app/controllers/external_controller.rb

```diff
@@ -29,6 +29,10 @@ def create_user
 
     # TODO: - Ahmad: deal with errors
 
+    redirect_location = cookies[:location]
+    cookies.delete(:location)
+    return redirect_to redirect_location if redirect_location&.match?('\/rooms\/\w{3}-\w{3}-\w{3}-\w{3}\/join')
```

Comment:
The regex isn't specific enough,
A redirection will be made to "https://www.google.com?ignore=/rooms/iam-abl-eto-pas/join" or anything similar,
However, rails `redirect_to` will catch that and raise `ActionController::Redirecting::UnsafeRedirectError` but we shouldn't be counting on that,

Reply:
Maybe `%r{\A/rooms/\w{3}-\w{3}-\w{3}-\w{3}/join\z}`?